### PR TITLE
cleanup and refactoring

### PR DESCRIPTION
**Pull request by niamster**
- integration with rubocop
- update according to the rubocop suggestions
- use facter to get the system info

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+AllCops:
```
Comment (tarcieri) [inline .rubocop.yml]: Would be nice if there were a consistent RuboCop style across all of the Celluloid projects

Perhaps another one for @digitalextremist's https://github.com/celluloid/culture/issues/2

Comment (niamster) [inline .rubocop.yml]: @tarcieri I see only that cellouloid enforces double-quote strings. The rubocop autogenerated config takes current state of the repo as a rule.
In DCell I've decided to have fine grained configuration. I would suggested the same thing in celluloid.
cc @digitalextremist

Comment (tarcieri) [inline .rubocop.yml]: Maybe we could have some overall stylistic guidelines for Celluloid projects as a whole and project-specific overrides

Comment (niamster) [inline .rubocop.yml]: I totally agree. It's even better to avoid additional per project configuration but share exactly same set of rules.

Comment (digitalextremist) [inline .rubocop.yml]: It seems like this `Exclude` list ought to make it into `culture/rubocop`

Comment (niamster) [inline .rubocop.yml]: @digitalextremist I believe all project files should be covered by rubocop. My intention was to step-by-step shrink that list and remove excludes entirely.

**Pull request merged**